crates/hyperswitch_domain_models/src/router_request_types.rs

```diff
@@ -546,14 +546,16 @@ impl
         let surcharge_amount = request_surcharge_details.surcharge_amount;
         let tax_on_surcharge_amount = request_surcharge_details.tax_amount.unwrap_or_default();
         Self {
-            original_amount: payment_attempt.amount,
+            original_amount: payment_attempt.net_amount.get_order_amount(),
             surcharge: common_utils::types::Surcharge::Fixed(
                 request_surcharge_details.surcharge_amount,
             ),
             tax_on_surcharge: None,
             surcharge_amount,
             tax_on_surcharge_amount,
-            final_amount: payment_attempt.amount + surcharge_amount + tax_on_surcharge_amount,
+            final_amount: payment_attempt.net_amount.get_order_amount()
```

Comment:
Why we are not considering `get_total_amount` here? Reason for again doing calculation here.

Reply:
We are trying to calculate surcharge_details from the RequestSurchargeDetails, net_amount is not computed yet. That's why we are getting surcharge details from RequestSurchargeDetails and calculating SurchargeDetails